---
title: "HW2"
author: "Gloria Chi"
date: "February 1, 2015"
output: html_document
---

Load libraries

```{r}
library(GEOmetadb)
library(data.table)
```

Download GEOmetadb database
```{r}
## This will download the entire database, so can be slow
if(!file.exists("GEOmetadb.sqlite"))
{
  # Download database only if it's not done already
  getSQLiteFile()
}
```
Connect to GEOmetadb
```{r}
geo_con <- dbConnect(SQLite(),'GEOmetadb.sqlite')
```

Figure out what components are in each table
```{r}
dbListTables(geo_con)
dbListFields(geo_con, 'gse')
dbListFields(geo_con, 'gpl')
dbListFields(geo_con, 'gsm')
dbListFields(geo_con, 'gse_gpl')
```

Query the database using the GEOmetabd package
```{r}
dbGetQuery(geo_con, "SELECT gse.title, gse.gse, gpl.gpl, gpl.manufacturer, gpl.description FROM (gse JOIN gse_gpl ON gse.gse=gse_gpl.gse) j JOIN gpl ON j.gpl=gpl.gpl WHERE (gse.summary LIKE '%HCV%' OR gse.title LIKE '%HCV%') AND gpl.manufacturer LIKE '%Illumina%' AND gse.contact LIKE '% Yale %'")
```

Convert all db tables to data.table tables
```{r}
gse.dt <-data.table(dbGetQuery(geo_con,"SELECT * FROM gse;"), key="gse")
gse_gpl.dt <-data.table(dbGetQuery(geo_con,"SELECT * FROM gse_gpl;"), key=c ("gse", "gpl"))
gpl.dt <-data.table(dbGetQuery(geo_con,"SELECT * FROM gpl;"), key="gpl")
```
Merge the data tables and query using data.table

```{r}

gpl.dt[gse_gpl.dt[gse.dt[(summary %like% 'HCV'| title %like%
'HCV') & contact %like% ' Yale ', list(gse, title)], list(gpl, i.gse,
title), nomatch=0], list(gpl, i.gse, title, manufacturer, description),
nomatch=0][manufacturer %like% 'Illumina']

```

Convert, merge, and subset all one line, but hard to read

```{r}
#Using brackets
(data.table(dbGetQuery(geo_con,"SELECT * FROM gpl;"), key="gpl"))[(data.table(dbGetQuery(geo_con,"SELECT * FROM gse_gpl;"), key=c ("gse", "gpl")))[(data.table(dbGetQuery(geo_con,"SELECT * FROM gse;"), key="gse"))[(summary %like% 'HCV'| title %like%
'HCV') & contact %like% ' Yale ', list(gse, title)], list(gpl, i.gse,
title), nomatch=0], list(gpl, i.gse, title, manufacturer, description),
nomatch=0][manufacturer %like% 'Illumina']

#Using merge
merge(data.table(dbGetQuery(geo_con,"SELECT gpl, manufacturer, description FROM gpl WHERE manufacturer LIKE '%Illumina%';"), key="gpl"), (merge(data.table(dbGetQuery(geo_con,"SELECT title, gse FROM gse WHERE contact LIKE '% Yale %' AND (summary LIKE '%HCV%' OR title LIKE '%HCV%');"), key="gse"), data.table(dbGetQuery(geo_con,"SELECT * FROM gse_gpl;"), key=c("gse", "gpl")), by="gse")), by="gpl")


```

```
##         gpl    i.gse                                        title
## 1: GPL10558 GSE40223 Illumina HumanHT-12 V4.0 expression beadchip
## 2: GPL10558 GSE40224 Illumina HumanHT-12 V4.0 expression beadchip
## 3: GPL10558 GSE40812 Illumina HumanHT-12 V4.0 expression beadchip
##     manufacturer
## 1: Illumina Inc.
## 2: Illumina Inc.
## 3: Illumina Inc.
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            description
## 1: The HumanHT-12 v4 Expression BeadChip provides high throughput processing of 12 samples per BeadChip without the need for expensive, specialized automation. The BeadChip is designed to support flexible usage across a wide-spectrum of experiments.;\t;\tThe updated content on the HumanHT-12 v4 Expression BeadChips provides more biologically meaningful results through genome-wide transcriptional coverage of well-characterized genes, gene candidates, and splice variants.;\t;\tEach array on the HumanHT-12 v4 Expression BeadChip targets more than 31,000 annotated genes with more than 47,000 probes derived from the National Center for Biotechnology Information Reference Sequence (NCBI) RefSeq Release 38 (November 7, 2009) and other sources.;\t;\tPlease use the GEO Data Submission Report Plug-in v1.0 for Gene Expression which may be downloaded from https://icom.illumina.com/icom/software.ilmn?id=234 to format the normalized and raw data.  These should be submitted as part of a GEOarchive.  Instructions for assembling a GEOarchive may be found at http://www.ncbi.nlm.nih.gov/projects/geo/info/spreadsheet.html;\t;\tOctober 11, 2012: annotation table updated with HumanHT-12_V4_0_R2_15002873_B.txt
## 2: The HumanHT-12 v4 Expression BeadChip provides high throughput processing of 12 samples per BeadChip without the need for expensive, specialized automation. The BeadChip is designed to support flexible usage across a wide-spectrum of experiments.;\t;\tThe updated content on the HumanHT-12 v4 Expression BeadChips provides more biologically meaningful results through genome-wide transcriptional coverage of well-characterized genes, gene candidates, and splice variants.;\t;\tEach array on the HumanHT-12 v4 Expression BeadChip targets more than 31,000 annotated genes with more than 47,000 probes derived from the National Center for Biotechnology Information Reference Sequence (NCBI) RefSeq Release 38 (November 7, 2009) and other sources.;\t;\tPlease use the GEO Data Submission Report Plug-in v1.0 for Gene Expression which may be downloaded from https://icom.illumina.com/icom/software.ilmn?id=234 to format the normalized and raw data.  These should be submitted as part of a GEOarchive.  Instructions for assembling a GEOarchive may be found at http://www.ncbi.nlm.nih.gov/projects/geo/info/spreadsheet.html;\t;\tOctober 11, 2012: annotation table updated with HumanHT-12_V4_0_R2_15002873_B.txt
## 3: The HumanHT-12 v4 Expression BeadChip provides high throughput processing of 12 samples per BeadChip without the need for expensive, specialized automation. The BeadChip is designed to support flexible usage across a wide-spectrum of experiments.;\t;\tThe updated content on the HumanHT-12 v4 Expression BeadChips provides more biologically meaningful results through genome-wide transcriptional coverage of well-characterized genes, gene candidates, and splice variants.;\t;\tEach array on the HumanHT-12 v4 Expression BeadChip targets more than 31,000 annotated genes with more than 47,000 probes derived from the National Center for Biotechnology Information Reference Sequence (NCBI) RefSeq Release 38 (November 7, 2009) and other sources.;\t;\tPlease use the GEO Data Submission Report Plug-in v1.0 for Gene Expression which may be downloaded from https://icom.illumina.com/icom/software.ilmn?id=234 to format the normalized and raw data.  These should be submitted as part of a GEOarchive.  Instructions for assembling a GEOarchive may be found at http://www.ncbi.nlm.nih.gov/projects/geo/info/spreadsheet.html;\t;\tOctober 11, 2012: annotation table updated with HumanHT-12_V4_0_R2_15002873_B.txt
```
